programs/Makefile

```diff
@@ -48,6 +48,7 @@ DEBUGFLAGS+=-Wall -Wextra -Wcast-qual -Wcast-align -Wshadow \
             -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings \
             -Wredundant-decls
 CFLAGS  += $(DEBUGFLAGS) $(MOREFLAGS)
+LDFLAGS += -rdynamic
```

Comment:
I'm unsure of the potential side effect of enabling this flag _always by default_.
I would suggest to keep it for debugging purpose,
hence document it (typically in `programs/README.md`).

Reply:
The only issue I am aware of would be more data in the symbol table. I will modify to use the `DEBUGFLAGS` option so it does not impact the `zstd-release` target. How would you want this documented?

Reply:
Even `DEBUGFLAGS` is too general, I'm afraid. 
`DEBUGFLAGS` is enabled for anything _but_ the `zstd-release` target.
However, `-rdynamic` is not compatible with MinGW for example. There could be more examples.
So automatically enabling this flag would require some advanced detection mechanisms, which would make this PR more complex. 

I believe the simpler solution would be to let this flag under user control. But it needs to be documented, for the user to be aware of it.

The way I see it, `programs/README.md` is the right place to document this flag, since it impacts the build process.
However, a paragraph must be added to document the new backtrace feature.